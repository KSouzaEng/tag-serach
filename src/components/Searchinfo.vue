<template>
<div id="serach">
    <h4></h4>
</div>
</template>

<script>
export default {

}
</script>

<style>

</style>