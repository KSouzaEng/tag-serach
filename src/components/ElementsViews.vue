<template>
  <div id="element">
    <div class="container">
        <div class="input-group flex-nowrap d-flex mt-5">
    <span class="input-group-text" id="addon-wrapping">@</span>
     <input type="text" class="form-control" placeholder="Username" aria-label="Username" aria-describedby="addon-wrapping">
    </div>
    </div>
  <Searchinfo  :word="word"/>

  </div>
</template>

<script>
import Searchinfo from './Searchinfo'

export default {
    name: 'ElementsViews',
    components:{
        Searchinfo,
    
    },
    // props:{
    //     word: String
    // },
    data(){
        return{
            word: ''
        }
    }
}
</script>

<style>

</style>