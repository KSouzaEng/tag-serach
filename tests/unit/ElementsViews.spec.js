import { shallowMount } from '@vue/test-utils'
import ElementsViews from '../../src/components/ElementsViews.vue'
import Searchinfo from '../../src/components/Searchinfo.vue'



describe('ElementsViews', () => {
    const build = () =>{
        const wrapper = shallowMount(ElementsViews)

        return{
            wrapper,
            info: () => wrapper.findComponent(Searchinfo)
            
        }
    }
    it('renders the component', () => {
 
        const { wrapper } = build()
      
        expect(wrapper.html()).toMatchSnapshot()
      })
      it('renders main child components',() => {

        const { info } = build()

        expect(info().exists()).toBe(true)

    })
    it('passes a binded user prop to Search component', () => {
        // arrange
        const { wrapper, info } = build()
        wrapper.setData({
          
            word: 'Karem'
          
        })
    
        expect(info().vm.word).toBe(wrapper.vm.word)
      })
  })