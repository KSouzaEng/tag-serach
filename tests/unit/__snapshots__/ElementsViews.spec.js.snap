// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementsViews renders the component 1`] = `
<div id="element">
  <div class="container">
    <div class="input-group flex-nowrap d-flex mt-5"><span class="input-group-text" id="addon-wrapping">@</span><input type="text" class="form-control" placeholder="Username" aria-label="Username" aria-describedby="addon-wrapping"></div>
  </div>
  <searchinfo-stub word=""></searchinfo-stub>
</div>
`;
